src/plugins/explore/public/components/query_panel/components/footer/query_error.tsx

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { i18n } from '@osd/i18n';
+
+import { EuiButtonEmpty, EuiPopover, EuiText, EuiPopoverTitle } from '@elastic/eui';
+
+import React, { useEffect, useMemo, useState } from 'react';
+import { ResultStatus } from '../../types';
+
+// TODO: Integrate this witih query error object once api itnregrated and make changes in error object and ui accoridnl
+export interface QueryStatus {
+  status: ResultStatus;
+  body?: {
+    error?: {
+      error?: string;
+      message?: {
+        error?:
+          | string
+          | {
+              reason?: string;
+              details: string;
+              type?: string;
+            };
+        status?: number;
+      };
+      statusCode?: number;
+    };
+  };
+  elapsedMs?: number;
+  startTime?: number;
+}
```

Comment:
i remember this type appeared somewhere else in OSD, does it need to be duplicated or should this be the only one in the future?